src/attest.ts

```diff
@@ -48,6 +58,48 @@ export const createAttestation = async (
 
     // Add the attestation's digest to the result
     result.attestationDigest = artifact.digest
+
+    // Because creating a storage record requires the 'artifact-metadata:write'
+    // permission, we wrap this in a try/catch to avoid failing the entire
+    // attestation process if the token does not have the correct permissions.
+    if (opts.createStorageRecord) {
+      try {
+        let subjectName = subject.name
+        const hasProtocol = /^[\w+.-]+:\/\//.test(subjectName)
+        const isHttps = subjectName.startsWith('https://')
+        if (hasProtocol && !isHttps) {
+          throw new Error(`Unsupported protocol in subject name`)
+        } else if (!hasProtocol) {
+          // if the subject name does not start with a protocol, prefix with "https://"
+          subjectName = `https://${subjectName}`
+        }
+        const registryUrl = new URL(subjectName).origin
+
+        const artifactOpts = {
+          name: subject.name,
+          digest: subjectDigest
+        }
+        const packageRegistryOpts = {
+          registryUrl
+        }
+        const records = await createStorageRecord(
+          artifactOpts,
+          packageRegistryOpts,
+          opts.githubToken
+        )
+
+        if (!records || records.length === 0) {
+          throw new Error('No storage records were created')
```

Comment:
I'm wondering if we really want to raise an error and cause the entire workflow to fail in  this case? If this were an explicit opt-in feature this might make sense, but users may get grumpy if their workflow fails due to an API error for a feature they didn't explicitly enable.

Should this just be logged as a warning?